tests/lib/linter/linter.js

```diff
@@ -1464,6 +1464,151 @@ describe("Linter", () => {
             assert.strictEqual(suppressedMessages.length, 0);
         });
 
+        it("rules use the rule's config when it is present", () => {
+            const config = {
+                rules: {
+                    "no-constant-condition": ["error", { checkLoops: "all" }]
+                }
+            };
+            const codeA = "/*eslint no-constant-condition: error */ while (true) {}";
+            const messages = linter.verify(codeA, config, filename, false);
```

Comment:
```suggestion
            const messages = linter.verify(codeA, config, filename);
```

`Linter#verify` accepts only 3 arguments.